### PR TITLE
Indicate where parsing failed

### LINKED ISSUE #17: Indicate which expression failed to parse
Currently, when an expression that the user passes in fails to parse, we don't provide any indication of _which_ box the parsing failed on. We simply return "false" when the user clicks the validation button, so the user can't even distinguish between a failure to parse and a failure of rule application. So we need to distinguish between those two errors in our output.

### PR DESCRIPTION
**Pull request by frogrammer**
Fix #17 
* Show a popup message indicating where parsing failed (they look ugly but that's a future task)
* `Validate` button doesn't work if there's an empty input box

### CODE REVIEW

File: src/edsger/core.cljs [hunk 1]
```diff
@@ -50,32 +56,85 @@
...
+        ;; vector of indices where parsing err occurred (e.g., [0, 3])
+        exp-parse-err (:locations (reduce merge-val {:curr-id -1 :locations []} exps))
+        rule-parse-err (:locations (reduce merge-val {:curr-id -1 :locations []} vanilla-rules))
         result-str (str (true? (uni/check-match-recursive (nth exps 0)
                                                           (nth exps 1)
                                                           (nth rules 0)
                                                           (nth rules 1))))]
-    (.alert js/window result-str)))
+    (when non-empty-input
+      (cond
```
Comment (lverns) [inline src/edsger/core.cljs]: The way this `cond` is set up, users have to resolve all the parse errors in the start & end expressions before the site tells them that there is anything wrong with the rules.

**Comment by frogrammer** [comment]
@lverns Do you think the expected behavior would be 1) clean up all existing error boxes 2) re-validate each box?

**Comment by lverns** [comment]
Yes, that sounds like the desired behavior.

File: src/edsger/core.cljs (hunk 1 above)
Comment (frogrammer) [reply src/edsger/core.cljs]: Fixed

**Comment by lverns** [comment]
I think this looks good.

**Pull request merged**